apps/portal-react/store/unitEpic.tsx

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable prettier/prettier */
+import { filter, switchMap, from, tap } from 'rxjs';
+import { requestUnits, loadedUnits } from './unitStore';
+import { Action } from 'redux';
+import { Observable, map } from 'rxjs';
+import { db } from '../firebase/database';
+import { collection, getDocs } from 'firebase/firestore';
+
+export const loadUnits = (action$: Observable<Action>) => 
+    action$.pipe(
+        filter((action) => action.type === requestUnits.type),
+        switchMap(() =>
+            from(getDocs(collection(db, 'units'))).pipe(
+                //tap((collection) => console.log(collection)),
+                map((collection) =>  loadedUnits(collection['_snapshot'].docChanges.map(unit => {
+                    
+                    const id = [...unit.doc.key.path.segments].pop()
```

Comment:
I don't know if we've upgraded the JS version yet, but eventually browsers will support `someArray.at(-1)` to get the last element (instead of having to copy the array and pop)